### PR TITLE
add generate example and simple shell command to execute the training

### PR DESCRIPTION
**Pull request by youkaichao**
This PR addresses the issue https://github.com/databrickslabs/dolly/issues/8

### CODE REVIEW

**Comment by srowen** [comment]
I don't think this belongs in a README - maybe just a separate .sh file in the repo? but I am also not clear if we want to duplicate the training code. .py code seems like the more natural delivery mechanism. Granted, it has shell commands in it, but Jupyter, Databricks et al can understand those. If anything I would like to just migrate the notebook to .ipynb

**Comment by youkaichao** [comment]
@srowen Hey, I can move the code into a separate .sh if you like. The problem this PR wants to resolve is that many people want to try Dolly without databricks environment, and a separate .sh would be super useful :)

**Comment by srowen** [comment]
I personally think the generate example is a good idea, but not a second copy of the training code, which seems more natural in Python. Databricks is not required, already.

File: generate.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+from transformers import AutoModelForCausalLM, AutoTokenizer
```
Comment (srowen) [inline generate.py:1]: Add space and copyright header

File: generate.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+path = '/path/to/model/checkpoint/directory'
+model_name = path
```
Comment (srowen) [inline generate.py:3]: this is redundant

File: generate.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+tokenizer = AutoTokenizer.from_pretrained(model_name)
+model = AutoModelForCausalLM.from_pretrained(model_name)
+
+import torch
+device = torch.device('cuda:0')
+model.to(device)
```
Comment (srowen) [inline generate.py:9]: just `model.to("cuda:0")` and inline it above

File: generate.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+def generate_text(model, tokenizer, input_text, device):
+    with torch.no_grad():
+        input_tokens = tokenizer.encode(input_text, return_tensors="pt").to(device)
+        output_tokens = model.generate(
+            input_tokens,
+            max_length=300,  # Controls the maximum length of the generated text
```
Comment (srowen) [inline generate.py:16]: Can we make these parameters that you can pass on the command line, so you can run with `python`?

**Review by srowen — CHANGES_REQUESTED** [review]
I'm not able to merge but i think these would be useful changes

**Comment by youkaichao** [comment]
I personally think the `train_dolly.py` is too long, and thus simplify it into several shell commands.

**Comment by srowen** [comment]
That's an advantage to being in a notebook, it's broken up naturally. I think you can maintain your own port as you like in a fork. The generate script sounds interesting though, that seems important to have.

**Comment by youkaichao** [comment]
Okay, then I will only keep the generate example.

**Comment by youkaichao** [comment]
By the way, the generate script is what I get from ChatGPT. Therefore, many hyper-parameters are not tuned :)

**Comment by srowen** [comment]
Oh I think @matthayes is going to add a generation script like what was used for the blog, to reproduce the result. That may already be coming

**Comment by matthayes** [comment]
Yes I just opened #27 for this.  Sorry for being late to post the generation code.
